pkgs/development/tools/protoc-gen-grpc-java/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchFromGitHub, protobuf }:
+
+stdenv.mkDerivation rec {
+  pname = "protoc-gen-grpc-java";
+  version = "1.50.2";
+
+  src = fetchFromGitHub {
+    owner = "grpc";
+    repo = "grpc-java";
+    rev = "v${version}";
+    sha256 = "sha256-06jBntoyYCQNNGr1hMgBGtc+tGYQUoPqHdTvHMm54rI=";
+  };
+
+  sourceRoot = "source/compiler/src/java_plugin/cpp";
+
+  patchPhase = ''
```

Comment:
```suggestion
  postPatch = ''
```

Reply:
I don't understand this suggestion. Why is `postPatch` preferred here, when I'm making the derivation from the ground up?

Reply:
because if you then add:

```
patches = [
  ./something.patch
  ./other-thing.patch
];
```

it wont work because you overwrote the `patchPhase`

and if you overwrite phases you should call the pre and post hooks yourself

Reply:
you reverted this